xarray/core/datatree.py

```diff
@@ -1607,3 +1616,35 @@ def to_zarr(
             compute=compute,
             **kwargs,
         )
+
+    def _get_all_dims(self) -> set:
```

Comment:
That sounds quite useful. Maybe this should be exposed as a public API (maybe a property)?

Reply:
I'm open to someone adding this later!

Reply:
I have to admit, that I never used DataTree... But what exactly does `DataTree.dims` return and how is it different to this?

Reply:
DataTree.dims returns all dimensions defined at the base level of the tree. This method also returns dimensions defined on descendant nodes.